File: backend/Api/Models/Inventory.cs
using System.Text.Json.Serialization;

namespace Api.Models
{
    public class Inventory
    {
        public int Id { get; set; }
        public int ProductId { get; set; }
        public int BranchId { get; set; }
        public int Quantity { get; set; }
        public int ReorderLevel { get; set; } = 10;
        public int MaxStockLevel { get; set; } = 100;
        public DateTime LastRestocked { get; set; } = DateTime.UtcNow;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        public virtual Product Product { get; set; } = null!;
        public virtual Branch Branch { get; set; } = null!;
    }
}

File: go-service/models/inventory.go
package models

import "time"

// Inventory represents stock levels at a branch
type Inventory struct {
	ID            int        `db:"Id" json:"id"`
	ProductID     int        `db:"ProductId" json:"productId"`
	BranchID      int        `db:"BranchId" json:"branchId"`
	Quantity      int        `db:"Quantity" json:"quantity"`
	ReorderLevel  int        `db:"ReorderLevel" json:"reorderLevel"`
	MaxStockLevel int        `db:"MaxStockLevel" json:"maxStockLevel"`
	LastRestocked time.Time  `db:"LastRestocked" json:"lastRestocked"`
	CreatedAt     time.Time  `db:"CreatedAt" json:"createdAt"`
	UpdatedAt     *time.Time `db:"UpdatedAt" json:"updatedAt,omitempty"`
}
